### PR TITLE
Feature/add mask field

### LINKED ISSUE #38: add a mask field to Geometry class
on creation `Geometry` should read in a land mask from a file (separate from what is being done in `State` now), specified by the yaml configuration and save the mask in an atlas field contained in `Geometry`

### PR DESCRIPTION
**Pull request by loganchen39**
@travissluka passed all test. I used the background file of "19850101_regridded_sst.nc" to create the landmask.nc, so they should be consistent. 

Close #38 .

### CODE REVIEW

File: src/umdsst/Geometry/Geometry.cc [hunk 1]
```diff
@@ -38,12 +51,71 @@ namespace umdsst {
 
     atlasFieldSet_.reset(new atlas::FieldSet());
     atlasFieldSet_->add(atlasFunctionSpace_->lonlat());
+
```
Comment (travissluka) [inline src/umdsst/Geometry/Geometry.cc]: the land mask will need to be copied as well in the copy constructor here.
I would remove the  above line with `lonlat` and just copy all the fields from `other->atlasFieldSet_` to this class' `atlasFieldSet_`

**Comment by travissluka** [comment]
It would also be very useful to print the number of masked and unmasked points in `Geometry::print()`

**Comment by travissluka** [comment]
I just double checked with the SOCA code, and the mask should be inverted. 1= ocean, 0=land. BUMP will expect 1 for the gridpoints that we want to used.

**Comment by loganchen39** [comment]
Ok, will update accordingly. Thanks,

File: src/umdsst/Geometry/Geometry.cc (hunk 1 above)
Comment (loganchen39) [reply src/umdsst/Geometry/Geometry.cc]: I used atlasFieldSet_-add() method to add (copy) all the fields from "other", hopefully it's ok.

**Comment by loganchen39** [comment]
@travissluka  I have updated accordingly. Please continue the CI check and merge the code if it's ok. Got distracted from the work.

**Comment by travissluka** [comment]
`staticbinit` is no longer passing, something apparently changed in saber that I missed. I'll try to fix that first

**Comment by loganchen39** [comment]
@travissluka  I've added your code from #52 , and it passes all tests. Please merge for now, and we'll update the code in the future if needed. 
Close #52 .

File: src/umdsst/Geometry/Geometry.cc [hunk 2]
```diff
@@ -37,13 +64,71 @@ namespace umdsst {
...
+      netCDF::NcVar varLandMask;
+      varLandMask = file.getVar("landmask");
+      if (varLandMask.isNull())
+        util::abor1_cpp("Get var landmask failed.", __FILE__, __LINE__);
+
+      int dataLandMask[lat][lon];
+      varLandMask.getVar(dataLandMask);
+
+      int idx = 0;
+      for (int j = 0; j < 180; j++)
```
Comment (travissluka) [inline src/umdsst/Geometry/Geometry.cc]: Can you remove the hardcoded 180 and 360 here and use the actual dimensions?

Comment (loganchen39) [reply src/umdsst/Geometry/Geometry.cc]: Just updated, thanks.

**Review by travissluka — APPROVED** [review]

**Pull request merged**